client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,19 +1,219 @@
+					original
+					isSmallestUnit
+				/>
+				<PlanPrice rawPrice={ introOfferMonthlyPrice } currencyCode={ currencyCode } />
+			</PricesGroup>
+		</>
+	);
+};
+
+interface PlanPriceOfferProps {
+	plan?: Plan;
+	currencyCode?: string;
+	originalMonthlyPrice?: number;
+	introOfferMonthlyPrice?: number;
+	formattedOriginalPrice?: string;
+	formattedIntroOfferPrice?: string;
+	introOfferEnabled: boolean;
+	isIntroOfferComplete: boolean;
+}
+
+const PlanPriceOffer = ( props: PlanPriceOfferProps ) => {
+	const translate = useTranslate();
+
+	const {
+		plan,
+		formattedOriginalPrice,
+		formattedIntroOfferPrice,
+		introOfferEnabled,
+		isIntroOfferComplete,
+		introOfferMonthlyPrice,
+		originalMonthlyPrice,
+		currencyCode,
+	} = props;
+
+	const showIntroOffer =
+		introOfferEnabled && ! isIntroOfferComplete && introOfferMonthlyPrice && originalMonthlyPrice;
+
+	const price = showIntroOffer ? (
+		<PriceWithIntroductoryOffer
+			originalMonthlyPrice={ originalMonthlyPrice }
+			introOfferMonthlyPrice={ introOfferMonthlyPrice }
+			currencyCode={ currencyCode }
+		/>
+	) : (
+		<PlanPrice rawPrice={ originalMonthlyPrice } currencyCode={ currencyCode } isSmallestUnit />
+	);
+
+	if ( plan && introOfferEnabled && ! isIntroOfferComplete ) {
+		plan.getBillingTimeFrame = () =>
+			translate(
+				'per month, %(discounted)s billed annually for the first year, %(original)s atferwards',
+				{
+					args: {
+						discounted: formattedIntroOfferPrice || '',
+						original: formattedOriginalPrice || '',
+					},
+				}
+			);
+	}
+
+	return (
+		<div className="import__upgrade-plan-price">
+			{ price }
+			<span className="plan-time-frame">
+				<small>{ plan?.getBillingTimeFrame() }</small>
+			</span>
+		</div>
+	);
+};
+
+const preparePlanPriceOfferProps = (
+	selectedPlan: string,
+	plan?: Plan,
+	pricing?: SitePlanPricing,
+	planDetails?: SitePlanData
+): PlanPriceOfferProps => {
+	let formattedOriginalPrice = undefined;
+	if ( planDetails && planDetails.formattedOriginalPrice ) {
+		formattedOriginalPrice = planDetails.formattedOriginalPrice;
+	}
+	if ( ! formattedOriginalPrice ) {
+		formattedOriginalPrice = planDetails?.formattedPrice || undefined;
+	}
+
+	const currencyCode = pricing?.currencyCode;
+	const originalMonthlyPrice = pricing?.originalPrice.monthly || undefined;
+
+	const introOfferRawPrice = pricing?.introOffer?.rawPrice;
+	const introOfferMonthlyPrice = introOfferRawPrice
+		? calculateMonthlyPriceForPlan( selectedPlan, introOfferRawPrice )
+		: undefined;
+	const formattedIntroOfferPrice = pricing?.introOffer?.formattedPrice || undefined;
+
+	const introOfferEnabled = config.isEnabled( 'migration-flow/introductory-offer' );
+	const isIntroOfferComplete = pricing?.introOffer?.isOfferComplete || false;
+
+	return {
+		plan,
+		currencyCode,
+		originalMonthlyPrice,
+		introOfferMonthlyPrice,
+		formattedOriginalPrice,
+		formattedIntroOfferPrice,
+		introOfferEnabled,
+		isIntroOfferComplete,
+	};
+};
```

Comment:
@daledupreez In case you start working on hooks.

Here I want to show that at the moment I had to use different sources (both Redux state and hooks) to gather all needed information.

I found that the hooks-approach has needed data (formatted original price) at the moment of getting response from API, but eliminates it while preparing the result.

Another interesting moment is that all sources have a mix of formats.

For example, we might have formatted price for the original price, but not for the introductory offer, or vice versa.
Or we have monthly and full price for original price and for discounted price (and they are smallest ints), but the introductory offer has only full and not the smallest int.
And so on.

I was very close to start changing the format, but realized it is a trap for me :)